src/components/user/AddFile.tsx

```diff
@@ -0,0 +1,226 @@
+import { createFile } from "@api/file/route.client";
+import { File as PrismaFile, Prisma } from "@prisma/client";
+import { Checkbox, button } from "@material-tailwind/react";
+import { setActive } from "@material-tailwind/react/components/Tabs/TabsContext";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCalendar } from "@fortawesome/free-solid-svg-icons";
+
+type AddFileProps = {
+  showAddFilePopUp: boolean;
+  setShowAddFilePopUp: Dispatch<SetStateAction<boolean>>;
+  seniorId: string;
+  files: PrismaFile[];
+  folder: string;
+};
+
+const TagOptions = ({
+  selectedTags,
+  setSelectedTags,
+}: {
+  selectedTags: TagProps[];
+  setSelectedTags: React.Dispatch<React.SetStateAction<TagProps[]>>;
+}) => {
+  return (
+    <div className="flex grid-flow-row flex-wrap">
+      {tagList.map((tag) => (
+        <div key={tag.name}>
+          <input
+            type="checkbox"
+            id={tag.name}
+            className="size-40 peer my-5 hidden"
+            onClick={() => {
+              if (!selectedTags.includes(tag) && selectedTags.length < 3) {
+                setSelectedTags([...selectedTags, tag]);
+              } else {
+                setSelectedTags(selectedTags.filter((i) => i != tag));
+              }
+            }}
+          />
+          {!selectedTags.includes(tag) && (
+            <label
+              htmlFor={tag.name}
+              className="m-2 ml-[-2px] inline-block rounded-full border-2 p-2 hover:bg-white hover:text-[#22555A]"
+            >
+              {" "}
+              {tag.name}
+            </label>
+          )}
+          {selectedTags.includes(tag) && (
+            <label
+              htmlFor={tag.name}
+              className="m-2 ml-[-2px] inline-block rounded-full border-2 bg-white p-2 text-[#22555A]"
+            >
+              {" "}
+              {tag.name}
+            </label>
+          )}
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const TagSelector = ({
+  selectedTags,
+  setSelectedTags,
+}: {
+  selectedTags: TagProps[];
+  setSelectedTags: React.Dispatch<React.SetStateAction<TagProps[]>>;
+}) => {
+  return (
+    <div>
+      <div className="text-neutral-600 mb-1 h-[34px] w-full font-['merriweather'] text-2xl">
+        Tags
+      </div>
+      <div className="text-lg font-thin">
+        Select min of 1, max of 3 <span className="text-red-400">*</span>
+      </div>
+      <TagOptions
+        selectedTags={selectedTags}
+        setSelectedTags={setSelectedTags}
+      />
+    </div>
+  );
+};
+
+const AddFile = ({
+  showAddFilePopUp,
+  setShowAddFilePopUp,
+  seniorId,
+  files,
+  folder,
+}: AddFileProps) => {
+  const currFiles = Object.values(files);
+  const excludeDates = currFiles.map((fileObj) => fileObj.date);
+  const excludedDatesString = excludeDates.map((dateObj) =>
+    dateObj.toDateString()
+  );
+
+  const startDate_ = new Date();
+
+  while (excludedDatesString.includes(startDate_.toDateString())) {
```

Comment:
nit: Instead of initializing `startDate_` here, you should do it in 

```ts
[startDate, setStartDate] = useState(() => // Your logic for start date here)
```

As an aside, let's check in to discuss another approach. Maybe we can set date as the current day, but not allow the user to create the file?